drivers/syslog/ramlog.c

```diff
@@ -423,24 +440,39 @@ static ssize_t ramlog_file_read(FAR struct file *filep, FAR char *buffer,
         }
       else
         {
+          /* Determine whether the read pointer is overwritten */
+
+          if (header->rl_head - upriv->rl_tail > priv->rl_bufsize)
+            {
+              upriv->rl_tail = header->rl_head - priv->rl_bufsize;
+            }
+
           /* The circular buffer is not empty, get the next byte from the
            * tail index.
            */
 
-          ch = header->rl_buffer[tail];
-
-          /* Increment the tail index. */
+          uint32_t pos = upriv->rl_tail % priv->rl_bufsize;
```

Comment:
need move before line 443

Reply:
upriv->rl_tail will update on line 447

Reply:
if so, we need separate the declaration and assignment.

Reply:
Done